L1Trigger/TrackFindingTracklet/src/TrackletProcessor.cc

```diff
@@ -374,51 +295,208 @@ void TrackletProcessor::execute() {
-                                                  << " " << outervmstubs_->nVMStubs()  //# outer stubs
-                                                  << " " << countteall                 //# pairs tried in TE
-                                                  << " " << stubpairs                  //# stubs pairs
-                                                  << " " << countsel                   //# tracklets found
-                                                  << endl;
+    bool notemptytebuffer = !tedatabuffer.empty();
+
+    int ite = -1;
+    for (auto& teunit : teunits_) {
+      ite++;
+      if (teunit.idle()) {
+        if (notemptytebuffer) {
+          teunit.init(std::get<0>(tebuffer_).read());
+          if (print)
+            std::cout << "istep=" << istep << " TEUnit init iTE inner : " << ite << " "
+                      << teunit.innerStub()->allStubIndex().value() << std::endl;
+          notemptytebuffer = false;  //prevent initialzing another TE unit
+        }
+      }
+      teunit.step(print, istep, ite);
     }
 
-    if (teunitsdone && tedatabuffersdone) {
+    //
+    // The third block here checks if we have input stubs to process
+    //
+    //
+      
+    if (goodtedata___)
+      tedatabuffer.store(tedata___);
+
+    goodtedata = false;
+
+    unsigned int& istub = std::get<1>(tebuffer_);
+    unsigned int& imem = std::get<2>(tebuffer_);
+    unsigned int imemend = std::get<4>(tebuffer_);
+
+    if ((!tebuffernearfull) && imem < imemend && istub < innerallstubs_[imem]->nStubs()) {
+      ninnerstubs++;
+
+      const Stub* stub = innerallstubs_[imem]->getStub(istub);
+      //const Stub* stub = innerallstubs_[imem]->getStub(innerallstubs_[imem]->nStubs()-istub-1);
+
+      if (settings_.debugTracklet()) {
+        edm::LogVerbatim("Tracklet") << getName() << " Have stub in " << innerallstubs_[imem]->getName();
+      }
+
+      bool negdisk = (stub->disk().value() < 0);  //FIXME stub needs to contain bit for +/- z disk
+
+      FPGAWord phicorr = stub->phicorr();
+      int innerfinephi = phicorr.bits(phicorr.nbits() - nbitsfinephi_, nbitsfinephi_);
+      FPGAWord innerbend = stub->bend();
+
+      //Take the top nbitszfinebintable_ bits of the z coordinate
+      int indexz = (stub->z().value() >> (stub->z().nbits() - nbitszfinebintable_)) & ((1 << nbitszfinebintable_) - 1);
+      int indexr = -1;
+      if (layerdisk1_ > (N_LAYER - 1)) {
+        if (negdisk) {
+          indexz = ((1 << nbitszfinebintable_) - 1) - indexz;
+        }
+        indexr = stub->r().value() >> (stub->r().nbits() - nbitsrfinebintable_);
+      } else {  //Take the top nbitsfinebintable_ bits of the z coordinate
+        indexr = (stub->r().value() >> (stub->r().nbits() - nbitsrfinebintable_)) & ((1 << nbitsrfinebintable_) - 1);
+      }
+
+      int lutval = -1;
+      if (iSeed_ < 6) {  //FIXME should only be one table...
```

Comment:
fix fixme?

Reply:
Not now - need non-trivial coordination with HLS development.